lib/Doctrine/DBAL/Driver/AbstractOracleDriver.php

```diff
@@ -118,20 +118,27 @@ protected function getEasyConnectString(array $params)
                 $params['port'] = 1521;
             }
 
-            $serviceName = $params['dbname'];
+            $serviceName = '';
 
             if ( ! empty($params['servicename'])) {
                 $serviceName = $params['servicename'];
+            } elseif(isset($params['dbname'])) {
+                $serviceName = $params['dbname'];
             }
 
-            $service = 'SID=' . $serviceName;
-            $pooled  = '';
-            $instance = '';
+            if (empty($serviceName)) {
+                throw new \InvalidArgumentException('No service name provided for Easy Connect String.');
+            }
 
             if (isset($params['service']) && $params['service'] == true) {
-                $service = 'SERVICE_NAME=' . $serviceName;
+                $service = '(SERVICE_NAME=' . $serviceName . ')';
+            } else {
+                $service = '(SID=' . $serviceName . ')';
```

Comment:
No need to add the parentheses here. Having to add them in multiple statements is cumbersome, when you can add them later on. Please leave it the way it was before.